core/src/conn.rs

```diff
@@ -43,6 +43,8 @@ impl MongoConnection {
         login_timeout: Option<u32>,
     ) -> Result<Self> {
         client_options.connect_timeout = login_timeout.map(|to| Duration::new(to as u64, 0));
+        client_options.max_pool_size = Some(1);
```

Comment:
The rust driver does implement connection pooling, all drivers do. Unfortunately, the default is 100, which is *way* too much and most people should reduce it from the default (knowledge I gained in previous role). I set it to 1 since we're not really doing anything that requires concurrent calls afaict. I'm happy to set it to something like 5 if we want.

Reply:
Edit, I set it to 5.